shared/packages/worker/src/worker/accessorHandlers/s3.ts

```diff
@@ -0,0 +1,608 @@
+		const endpoint = this.accessor.endpoint
+		const forcePathStyle = this.accessor.forcePathStyle
+
+		if (accessKey === undefined) throw new Error('S3AccessorHandle: accessKey is not set!')
+		if (secretAccessKey === undefined) throw new Error('S3AccessorHandle: secretAccessKey is not set!')
+		if (bucketId === undefined) throw new Error('S3AccessorHandle: bucketId is not set!')
+		if (region === undefined) throw new Error('S3AccessorHandle: region is not set!')
+		if (s3PublicBaseUrl === undefined) throw new Error('S3AccessorHandle: s3PublicBaseUrl is not set!')
+
+		return {
+			accessKey,
+			secretAccessKey,
+			bucketId,
+			region,
+			s3PublicBaseUrl,
+			endpoint,
+			forcePathStyle,
+		}
+	}
+
+	private async getS3Client(): Promise<S3BucketClient> {
+		let client = this.getS3ClientFromCache(
+			this.s3Options.region,
+			this.s3Options.accessKey,
+			this.s3Options.secretAccessKey,
+			this.s3Options.s3PublicBaseUrl,
+			this.s3Options.endpoint,
+			this.s3Options.forcePathStyle
+		)
+
+		if (!client) {
+			client = this.createNewS3Client()
+
+			this.cacheS3Client(
+				client,
+				this.s3Options.region,
+				this.s3Options.accessKey,
+				this.s3Options.secretAccessKey,
+				this.s3Options.s3PublicBaseUrl,
+				this.s3Options.endpoint,
+				this.s3Options.forcePathStyle
+			)
+		}
+
+		return client
+	}
+
+	private getS3ClientFromCache(
+		region: string,
+		accessKey: string,
+		secretAccessKey: string,
+		publicBaseUrl: string,
+		endpoint?: string,
+		forcePathStyle?: boolean
+	): S3BucketClient | undefined {
+		const accessorCache = this.worker.accessorCache as {
+			[accessorType: string]: S3BucketClient | undefined
+		}
+
+		return accessorCache[
+			S3AccessorHandle.getCacheKeyForS3Client(
+				accessKey,
+				secretAccessKey,
+				region,
+				publicBaseUrl,
+				endpoint,
+				forcePathStyle
+			)
+		]
+	}
+
+	private cacheS3Client(
+		client: S3BucketClient,
+		region: string,
+		accessKey: string,
+		secretAccessKey: string,
+		publicBaseUrl: string,
+		endpoint?: string,
+		forcePathStyle?: boolean
+	): void {
+		const accessorCache = this.worker.accessorCache as {
+			[accessorType: string]: S3BucketClient | undefined
+		}
+
+		accessorCache[
+			S3AccessorHandle.getCacheKeyForS3Client(
+				accessKey,
+				secretAccessKey,
+				region,
+				publicBaseUrl,
+				endpoint,
+				forcePathStyle
+			)
+		] = client
+	}
+
+	static getCacheKeyForS3Client(
+		accessKey: string,
+		secretAccessKey: string,
+		region: string,
+		publicBaseUrl: string,
+		endpoint?: string,
+		forcePathStyle?: boolean
+	): string {
+		return JSON.stringify([accessKey, secretAccessKey, region, publicBaseUrl, endpoint, forcePathStyle])
+	}
+
+	private createNewS3Client(): S3BucketClient {
+		return new S3BucketClient(
+			this.s3Options.bucketId,
+			this.s3Options.region,
+			this.s3Options.accessKey,
+			this.s3Options.secretAccessKey,
+			this.s3Options.endpoint,
+			this.s3Options.forcePathStyle
+		)
+	}
```

Comment:
This can be simplified using `worker.cacheData()`.

```suggestion
		private async getS3Client(): Promise<S3BucketClient> {
		const cacheKey = JSON.stringify(this.s3Options)

		return this.worker.cacheData(
			this.type,
			`s3Client${cacheKey}`,
			() => {
				return this.createNewS3Client()
			},
			10 * 60 * 1000
		) // 10 min cache time
	}

	private createNewS3Client(): S3BucketClient {
		return new S3BucketClient(this.s3Options)
	}
```